pikaur_test/test_sysupgrade.py

```diff
@@ -75,9 +75,9 @@ def test_devel_upgrade(self):
 
         self.downgrade_dev_pkg()
 
-        query_result = pikaur('-Qu').stdout
+        query_result = pikaur('-Qu').stdout[:-1]  # remove \n
```

Comment:
```
>>> '\n'[:-1].splitlines()
[]

>>> 'foo\nbar\n'[:-1].splitlines()
['foo', 'bar']

>>> ''[:-1].splitlines()
[]
```
I don't think the problem is the empty line... at least not anymore. It is more of comparing result string with `\n` at the end against the test string without `\n`. Because you are using test string everywhere, I added more primitive `[:-1]` here instead of `strip()`.

The alternative might be to test `string + '\n'` or `f'{string}\n'`. This passes the test too.

Reply:
so apply splitlines to result and compare it after splitting

Reply:
Just this? Leave others alone? 
```diff
diff --git a/pikaur_test/test_sysupgrade.py b/pikaur_test/test_sysupgrade.py
index b539a20..ff78c28 100644
--- a/pikaur_test/test_sysupgrade.py
+++ b/pikaur_test/test_sysupgrade.py
@@ -75,7 +75,7 @@ class SysupgradeTest(PikaurDbTestCase):
 
         self.downgrade_dev_pkg()
 
-        query_result = pikaur('-Qu').stdout[:-1]  # remove \n
+        query_result = pikaur('-Qu').stdout
         self.assertEqual(
             len(query_result.splitlines()), 0
         )
```

Reply:
I wanted to clear up the confusion. I am explaining all diffs. Are you satisfied with `[:-1]` or do you prefer clearer alternatives such as testing `string + '\n'` or `f'{string}\n'`? With no empty line, `splitlines()` should work here, but not others.